### ISSUE TITLE
Incremental update failing infinitely

### ISSUE THREAD

**Issue by LarnuUK** [Cisco-Talos/clamav#217]
Describe the bug
----------------
I cannot update the database version for Clam AV. I effectively get stuck in an infinite loop where the application downloads the update, errors, downloads the update, errors, downloads the update, errors... This is a problem as, as the moment, I have to have the `clamav-freshclam` service permanently disabled as otherwise my internet is unusable while my PC is switched on (due the whole of by pitiful 10Mbps being abused by Clam AV).

This looks like the same as this [issue](https://github.com/Cisco-Talos/clamav/issues/201), however, that states to open a new ticket if the problem is still occurring and I am using 0.103.2 (latest version available via the Ubuntu Package Manager).

How to reproduce the problem
----------------------------
Leave `clamav-freshclam` service and it will continually try to update and error. Running `sudo freshclam` gives the following:
```none
WARNING: Ignoring deprecated option SafeBrowsing at /etc/clamav/freshclam.conf:22
Tue Jul 20 09:52:30 2021 -> ClamAV update process started at Tue Jul 20 09:52:30 2021
Tue Jul 20 09:52:30 2021 -> ^Your ClamAV installation is OUTDATED!
Tue Jul 20 09:52:30 2021 -> ^Local version: 0.103.2 Recommended version: 0.103.3
Tue Jul 20 09:52:30 2021 -> DON'T PANIC! Read https://www.clamav.net/documents/upgrading-clamav
Tue Jul 20 09:52:30 2021 -> ^Cool-down expired, ok to try again.
Tue Jul 20 09:52:30 2021 -> daily database available for update (local version: 26231, remote version: 26238)
Current database is 7 versions behind.
Downloading database patch # 26232...
Tue Jul 20 09:52:30 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 09:52:30 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 09:52:30 2021 -> ^Incremental update failed, trying to download daily.cvd
Time:   58.5s, ETA:    0.0s [========================>]   54.81MiB/54.81MiB
Tue Jul 20 09:53:30 2021 -> Received an older daily CVD than was advertised. We'll retry so the incremental update will ensure we're up-to-date.
Tue Jul 20 09:53:30 2021 -> daily database available for update (local version: 26231, remote version: 26238)
Current database is 7 versions behind.
Downloading database patch # 26232...
Tue Jul 20 09:53:30 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 09:53:30 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 09:53:30 2021 -> ^Incremental update failed, trying to download daily.cvd
Time:   58.2s, ETA:    0.0s [========================>]   54.81MiB/54.81MiB
Tue Jul 20 09:54:29 2021 -> Received an older daily CVD than was advertised. We'll retry so the incremental update will ensure we're up-to-date.
Tue Jul 20 09:54:29 2021 -> daily database available for update (local version: 26231, remote version: 26238)
Current database is 7 versions behind.
Downloading database patch # 26232...
Tue Jul 20 09:54:30 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 09:54:30 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 09:54:30 2021 -> ^Incremental update failed, trying to download daily.cvd
Time:   59.5s, ETA:    0.0s [========================>]   54.81MiB/54.81MiB
Tue Jul 20 09:55:30 2021 -> Received an older daily CVD than was advertised. We'll retry so the incremental update will ensure we're up-to-date.
Tue Jul 20 09:55:30 2021 -> daily database available for update (local version: 26231, remote version: 26238)
Current database is 7 versions behind.
Downloading database patch # 26232...
Tue Jul 20 09:55:30 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 09:55:30 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 09:55:30 2021 -> ^Incremental update failed, trying to download daily.cvd
Time:   59.8s, ETA:    0.0s [========================>]   54.81MiB/54.81MiB
Tue Jul 20 09:56:31 2021 -> Received an older daily CVD than was advertised. We'll retry so the incremental update will ensure we're up-to-date.
Tue Jul 20 09:56:31 2021 -> daily database available for update (local version: 26231, remote version: 26238)
Current database is 7 versions behind.
Downloading database patch # 26232...
Tue Jul 20 09:56:31 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 09:56:31 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 09:56:31 2021 -> ^Incremental update failed, trying to download daily.cvd
Time:   59.5s, ETA:    0.0s [========================>]   54.81MiB/54.81MiB
Tue Jul 20 09:57:32 2021 -> Received an older daily CVD than was advertised. We'll retry so the incremental update will ensure we're up-to-date.
Tue Jul 20 09:57:32 2021 -> daily database available for update (local version: 26231, remote version: 26238)
Current database is 7 versions behind.
Downloading database patch # 26232...
Tue Jul 20 09:57:32 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 09:57:32 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 09:57:32 2021 -> ^Incremental update failed, trying to download daily.cvd
Time:   59.0s, ETA:    0.0s [========================>]   54.81MiB/54.81MiB
Tue Jul 20 09:58:32 2021 -> Received an older daily CVD than was advertised. We'll retry so the incremental update will ensure we're up-to-date.
Tue Jul 20 09:58:32 2021 -> daily database available for update (local version: 26231, remote version: 26238)
Current database is 7 versions behind.
Downloading database patch # 26232...
Tue Jul 20 09:58:33 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 09:58:33 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 09:58:33 2021 -> ^Incremental update failed, trying to download daily.cvd
^CTue Jul 20 09:58:46 2021 -> Update process terminated   12.78MiB/54.81MiB
```

Replace this text with the output from the ClamAV command:
```none
Checking configuration files in /etc/clamav

Config file: clamd.conf
-----------------------
PreludeAnalyzerName = "ClamAV"
LogFile = "/var/log/clamav/clamav.log"
LogFileMaxSize = "4294967295"
LogTime = "yes"
LogRotate = "yes"
ExtendedDetectionInfo = "yes"
LocalSocket = "/var/run/clamav/clamd.ctl"
LocalSocketGroup = "clamav"
LocalSocketMode = "666"
MaxConnectionQueueLength = "15"
MaxThreads = "12"
ReadTimeout = "180"
SendBufTimeout = "200"
SelfCheck = "3600"
User = "clamav"
BytecodeTimeout = "60000"
MaxScanTime = "120000"
PCREMatchLimit = "10000"
PCRERecMatchLimit = "5000"

Config file: freshclam.conf
---------------------------
LogFileMaxSize = "4294967295"
LogTime = "yes"
LogRotate = "yes"
UpdateLogFile = "/var/log/clamav/freshclam.log"
Checks = "24"
DatabaseMirror = "db.local.clamav.net", "database.clamav.net"
MaxAttempts = "5"
ConnectTimeout = "90"
ReceiveTimeout = "90"
*** SafeBrowsing is DEPRECATED ***

clamav-milter.conf not found

Software settings
-----------------
Version: 0.103.2
Optional features supported: MEMPOOL IPv6 FRESHCLAM_DNS_FIX AUTOIT_EA06 BZIP2 LIBXML2 PCRE2 ICONV JSON 

Database information
--------------------
Database directory: /var/lib/clamav
daily.cld: version 26231, sigs: 3996055, built on Wed Jul 14 12:05:45 2021
bytecode.cld: version 333, sigs: 92, built on Mon Mar  8 15:21:51 2021
main.cld: version 59, sigs: 4564902, built on Mon Nov 25 13:56:15 2019
Total number of signatures: 8561049

Platform information
--------------------
uname: Linux 5.4.0-77-generic #86-Ubuntu SMP Thu Jun 17 02:35:03 UTC 2021 x86_64
OS: linux-gnu, ARCH: x86_64, CPU: x86_64
Full OS version: Ubuntu 20.04.2 LTS
zlib version: 1.2.11 (1.2.11), compile flags: a9
platform id: 0x0a217b7b0800000000090300

Build information
-----------------
GNU C: 9.3.0 (9.3.0)
CPPFLAGS: -Wdate-time -D_FORTIFY_SOURCE=2
CFLAGS: -g -O2 -fdebug-prefix-map=/build/clamav-TW4JTf/clamav-0.103.2+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -D_FILE_OFFSET_BITS=64  -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
CXXFLAGS: -g -O2 -fdebug-prefix-map=/build/clamav-TW4JTf/clamav-0.103.2+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -D_FILE_OFFSET_BITS=64
LDFLAGS: -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -Wl,--as-needed
Configure: '--build=x86_64-linux-gnu' '--prefix=/usr' '--includedir=/usr/include' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--sysconfdir=/etc' '--localstatedir=/var' '--disable-silent-rules' '--libdir=/usr/lib/x86_64-linux-gnu' '--runstatedir=/run' '--disable-maintainer-mode' '--disable-dependency-tracking' 'CFLAGS=-g -O2 -fdebug-prefix-map=/build/clamav-TW4JTf/clamav-0.103.2+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -D_FILE_OFFSET_BITS=64' 'CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2' 'CXXFLAGS=-g -O2 -fdebug-prefix-map=/build/clamav-TW4JTf/clamav-0.103.2+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security -Wall -D_FILE_OFFSET_BITS=64' 'LDFLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,now -Wl,--as-needed' '--with-dbdir=/var/lib/clamav' '--sysconfdir=/etc/clamav' '--disable-clamav' '--disable-unrar' '--enable-milter' '--enable-dns-fix' '--with-libjson' '--with-system-libmspack' '--with-libcurl=/usr' '--with-gnu-ld' '--with-systemdsystemunitdir=/lib/systemd/system' 'build_alias=x86_64-linux-gnu' 'OBJCFLAGS=-g -O2 -fdebug-prefix-map=/build/clamav-TW4JTf/clamav-0.103.2+dfsg=. -fstack-protector-strong -Wformat -Werror=format-security'
sizeof(void*) = 8
Engine flevel: 123, dconf: 123

```

Attachments
-----------
None

**Comment by micahsnyder** [comment]
Hi @LarnuUK,.
This report is a little startling to me.  I haven't seen it before.  I will try to reproduce the issue.  I believe this should not happen, at least not in clamav 0.103.3 (I see you're on 0.103.2).  I hope others aren't being affects as well.  

You should be able to resolve this issue by deleting `/var/lib/clamav/daily.cld` and then trying again. 

Side note: There are some other issues with freshclam made apparent by the recent major update to daily.cvd and main.cvd (the issue causing this `cdiff_apply: lseek(desc, -350, SEEK_END) failed` error message). We will have to fix the rest of those issues in 0.103.4 & 0.104.1, after the 0.104.0 release.

**Comment by LarnuUK** [comment]
Thanks @micahsnyder . Unfortunately, I have been rate limited _again_ (I feel this a little unfair as it's not my fault by the application's) so I can't test this for a while.

It would be very useful if ClamAV didn't cause the rate limit problem when it is errorring; either by not trying again after couple of failed attempts or because it's clear that the application is at fault the rate limit isn't trigger on the server.

**Comment by finchy** [comment]
I will disable the rate limit for a while.

**Comment by micahsnyder** [comment]
@LarnuUK give it a try now, hopefully we can get you up and running again.

**Comment by LarnuUK** [comment]
Side Note, in regards to using 0.103.2 it appears this is the latest version available in the Ubuntu Repository. Before posting the bug I did try forcing an update (`sudo apt install clam-av`) which advised that the I'm using the latest version (0.103.2+dfsg-0ubuntu0.20.04.2).

> @LarnuUK give it a try now, hopefully we can get you up and running again.

Thanks I gave it a go, but it says I'm still on cool down till 09:58:12 (BST) tomorrow.

**Comment by micahsnyder** [comment]
@LarnuUK that's freshclam trying to blindly follow the rate limiting request to prevent extra rate limiting, even though @finchy lifted the requirement for now.  Please try to delete the `.dat` file (mirrors.dat or freshclam.dat, depending on your version) in `/var/lib/clamav/` and try again.

**Comment by LarnuUK** [comment]
Thanks both @micahsnyder  and @finchy .

After deleting `daily.cld` and renaming `mirrors.dat`, `frfeshclam` updated `daily.cvd` It then tried to download `main.cvd` but failed, due to the size and my slow internet, 10Mbps~ (I honestly cannot wait till OpenReach finally install Fibre for me; COVID delayed it by well over a year), it timed out after 90 seconds.

I therefore then upped the value of `ReceiveTimeout` in `/etc/clamav/freshclam.conf` to 300 and ran `freshclam` again. This then updated the application in full. Awesome!

I do wonder if due to the large new file that needed to be downloaded this was a contributing factor. I noticed in the linked issue that that user too had limited bandwidth. Although I will admit it's expected that most people have internet speeds of around 50mbps+ unfortunately not all of us do, and a "small" download of 180MB~ can therefore take several minutes. Wonder if the default value for `ReceiveTimeout` should be higher?

I've included the output from bash below anyway, as there were a couple of error, but I think you cover this, micah, in your first comment's side note.

Many thanks both again.

Bash output:
```bash
thom@neith:/var/lib/clamav$ sudo freshclam 
WARNING: Ignoring deprecated option SafeBrowsing at /etc/clamav/freshclam.conf:22
Tue Jul 20 22:25:46 2021 -> ClamAV update process started at Tue Jul 20 22:25:46 2021
Tue Jul 20 22:25:46 2021 -> ^Your ClamAV installation is OUTDATED!
Tue Jul 20 22:25:46 2021 -> ^Local version: 0.103.2 Recommended version: 0.103.3
Tue Jul 20 22:25:46 2021 -> DON'T PANIC! Read https://www.clamav.net/documents/upgrading-clamav
Tue Jul 20 22:25:46 2021 -> daily database available for download (remote version: 26238)
Time:   54.6s, ETA:    0.0s [========================>]   54.83MiB/54.83MiB
Tue Jul 20 22:26:42 2021 -> Testing database: '/var/lib/clamav/tmp.4af3f96a91/clamav-e3c382c51c7b89d08e4ccbf8e705e220.tmp-daily.cvd' ...
Tue Jul 20 22:26:46 2021 -> Database test passed.
Tue Jul 20 22:26:46 2021 -> daily.cvd updated (version: 26238, sigs: 1962783, f-level: 90, builder: raynman)
Tue Jul 20 22:26:46 2021 -> main database available for update (local version: 59, remote version: 61)
Current database is 2 versions behind.
Downloading database patch # 60...
Tue Jul 20 22:26:46 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 22:26:46 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 22:26:46 2021 -> ^Incremental update failed, trying to download main.cvd
Tue Jul 20 22:28:16 2021 -> ^Download failed (28) Tue Jul 20 22:28:16 2021 -> ^ Message: Timeout was reached
Tue Jul 20 22:28:16 2021 -> ^Can't download main.cvd from https://database.clamav.net/main.cvd
Tue Jul 20 22:28:16 2021 -> Trying again in 5 secs...
^CTue Jul 20 22:28:18 2021 -> Update process terminated
thom@neith:/var/lib/clamav$ sudo nano /etc/clamav/freshclam.conf 
thom@neith:/var/lib/clamav$ sudo freshclam 
WARNING: Ignoring deprecated option SafeBrowsing at /etc/clamav/freshclam.conf:22
Tue Jul 20 22:31:17 2021 -> ClamAV update process started at Tue Jul 20 22:31:17 2021
Tue Jul 20 22:31:17 2021 -> ^Your ClamAV installation is OUTDATED!
Tue Jul 20 22:31:17 2021 -> ^Local version: 0.103.2 Recommended version: 0.103.3
Tue Jul 20 22:31:17 2021 -> DON'T PANIC! Read https://www.clamav.net/documents/upgrading-clamav
Tue Jul 20 22:31:17 2021 -> daily.cvd database is up-to-date (version: 26238, sigs: 1962783, f-level: 90, builder: raynman)
Tue Jul 20 22:31:17 2021 -> main database available for update (local version: 59, remote version: 61)
Current database is 2 versions behind.
Downloading database patch # 60...
Tue Jul 20 22:31:17 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 22:31:17 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 22:31:17 2021 -> ^Incremental update failed, trying to download main.cvd
Time:  2m 23s, ETA:    0.0s [========================>]  160.41MiB/160.41MiB
Tue Jul 20 22:33:43 2021 -> Testing database: '/var/lib/clamav/tmp.258f69a770/clamav-caa5efbbe0daf7024e2f2bc4f1f4c9d6.tmp-main.cvd' ...
Tue Jul 20 22:33:48 2021 -> Database test passed.
Tue Jul 20 22:33:48 2021 -> main.cvd updated (version: 61, sigs: 6607162, f-level: 90, builder: sigmgr)
Tue Jul 20 22:33:48 2021 -> bytecode.cld database is up-to-date (version: 333, sigs: 92, f-level: 63, builder: awillia2)
Tue Jul 20 22:33:48 2021 -> Clamd successfully notified about the update.
```

**Comment by finchy** [comment]
Yeah. Probably not a bad thought.  Maybe we should post a blog post about that as well.

—
Sent from my  iPhone

On Jul 20, 2021, at 17:38, Thom A ***@***.***> wrote:

﻿

Thanks both @micahsnyder<https://github.com/micahsnyder> and @finchy<https://github.com/finchy> .

After deleting daily.cld and renaming mirrors.dat, frfeshclam updated daily.cvd It then tried to download main.cvd but failed, due to the size and my slow internet, 10Mbps~ (I honestly cannot wait till OpenReach finally install Fibre for me; COVID delayed it by well over a year), it timed out after 90 seconds.

I therefore then upped the value of ReceiveTimeout in /etc/clamav/freshclam.conf to 300 and ran freshclam again. This then updated the application in full. Awesome!

I do wonder if due to the large new file that needed to be downloaded this was a contributing factor. I noticed in the linked issue that that user too had limited bandwidth. Although I will admit it's expected that most people have internet speeds of around 50mbps+ unfortunately not all of us do, and a "small" download of 180MB~ can therefore take several minutes. Wonder if the default value for ReceiveTimeout should be higher?

I've included the output from bash below anyway, as there were a couple of error, but I think you cover this, micah, in your first comment's side note.

Many thanks both again.

Bash output:

***@***.***:/var/lib/clamav$ sudo freshclam
WARNING: Ignoring deprecated option SafeBrowsing at /etc/clamav/freshclam.conf:22
Tue Jul 20 22:25:46 2021 -> ClamAV update process started at Tue Jul 20 22:25:46 2021
Tue Jul 20 22:25:46 2021 -> ^Your ClamAV installation is OUTDATED!
Tue Jul 20 22:25:46 2021 -> ^Local version: 0.103.2 Recommended version: 0.103.3
Tue Jul 20 22:25:46 2021 -> DON'T PANIC! Read https://www.clamav.net/documents/upgrading-clamav
Tue Jul 20 22:25:46 2021 -> daily database available for download (remote version: 26238)
Time:   54.6s, ETA:    0.0s [========================>]   54.83MiB/54.83MiB
Tue Jul 20 22:26:42 2021 -> Testing database: '/var/lib/clamav/tmp.4af3f96a91/clamav-e3c382c51c7b89d08e4ccbf8e705e220.tmp-daily.cvd' ...
Tue Jul 20 22:26:46 2021 -> Database test passed.
Tue Jul 20 22:26:46 2021 -> daily.cvd updated (version: 26238, sigs: 1962783, f-level: 90, builder: raynman)
Tue Jul 20 22:26:46 2021 -> main database available for update (local version: 59, remote version: 61)
Current database is 2 versions behind.
Downloading database patch # 60...
Tue Jul 20 22:26:46 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 22:26:46 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 22:26:46 2021 -> ^Incremental update failed, trying to download main.cvd
Tue Jul 20 22:28:16 2021 -> ^Download failed (28) Tue Jul 20 22:28:16 2021 -> ^ Message: Timeout was reached
Tue Jul 20 22:28:16 2021 -> ^Can't download main.cvd from https://database.clamav.net/main.cvd
Tue Jul 20 22:28:16 2021 -> Trying again in 5 secs...
^CTue Jul 20 22:28:18 2021 -> Update process terminated
***@***.***:/var/lib/clamav$ sudo nano /etc/clamav/freshclam.conf
***@***.***:/var/lib/clamav$ sudo freshclam
WARNING: Ignoring deprecated option SafeBrowsing at /etc/clamav/freshclam.conf:22
Tue Jul 20 22:31:17 2021 -> ClamAV update process started at Tue Jul 20 22:31:17 2021
Tue Jul 20 22:31:17 2021 -> ^Your ClamAV installation is OUTDATED!
Tue Jul 20 22:31:17 2021 -> ^Local version: 0.103.2 Recommended version: 0.103.3
Tue Jul 20 22:31:17 2021 -> DON'T PANIC! Read https://www.clamav.net/documents/upgrading-clamav
Tue Jul 20 22:31:17 2021 -> daily.cvd database is up-to-date (version: 26238, sigs: 1962783, f-level: 90, builder: raynman)
Tue Jul 20 22:31:17 2021 -> main database available for update (local version: 59, remote version: 61)
Current database is 2 versions behind.
Downloading database patch # 60...
Tue Jul 20 22:31:17 2021 -> !cdiff_apply: lseek(desc, -350, SEEK_END) failed
Tue Jul 20 22:31:17 2021 -> !downloadPatch: Can't apply patch
Tue Jul 20 22:31:17 2021 -> ^Incremental update failed, trying to download main.cvd
Time:  2m 23s, ETA:    0.0s [========================>]  160.41MiB/160.41MiB
Tue Jul 20 22:33:43 2021 -> Testing database: '/var/lib/clamav/tmp.258f69a770/clamav-caa5efbbe0daf7024e2f2bc4f1f4c9d6.tmp-main.cvd' ...
Tue Jul 20 22:33:48 2021 -> Database test passed.
Tue Jul 20 22:33:48 2021 -> main.cvd updated (version: 61, sigs: 6607162, f-level: 90, builder: sigmgr)
Tue Jul 20 22:33:48 2021 -> bytecode.cld database is up-to-date (version: 333, sigs: 92, f-level: 63, builder: awillia2)
Tue Jul 20 22:33:48 2021 -> Clamd successfully notified about the update.

**Comment by micahsnyder** [comment]
@LarnuUK 

> After deleting `daily.cld` and renaming `mirrors.dat`, `freshclam` updated `daily.cvd` It then tried to download `main.cvd` but failed, due to the size and my slow internet, 10Mbps~ (I honestly cannot wait till OpenReach finally install Fibre for me; COVID delayed it by well over a year), it timed out after 90 seconds.
> 
> I therefore then upped the value of `ReceiveTimeout` in `/etc/clamav/freshclam.conf` to 300 and ran `freshclam` again. This then updated the application in full. Awesome!
> 
> I do wonder if due to the large new file that needed to be downloaded this was a contributing factor. I noticed in the linked issue that that user too had limited bandwidth. Although I will admit it's expected that most people have internet speeds of around 50mbps+ unfortunately not all of us do, and a "small" download of 180MB~ can therefore take several minutes. Wonder if the default value for `ReceiveTimeout` should be higher?

ClamAV's default ReceiveTimeout is `0` (no timeout).  The setting in your config came from Ubuntu's packaging choices.  We should get Ubuntu & Debian and others to remove this option from their configs. 

The timeout definitely may be a contributing factor causing the infinite loop.  I will have to investigate that when I test.  I'm presently bogged down with 0.104.0 release-candidate preparation, trying to get ready to publish it tomorrow. I'll update the ticket if I learn anything new.

**Comment by micahsnyder** [comment]
@finchy 

> Yeah. Probably not a bad thought. Maybe we should post a blog post about that as well. — Sent from my  iPhone On Jul 20, 

A blog post is a good idea.

**Comment by mbohde** [comment]
> Hi @LarnuUK,.
> This report is a little startling to me. I haven't seen it before. I will try to reproduce the issue. I believe this should not happen, at least not in clamav 0.103.3 (I see you're on 0.103.2). I hope others aren't being affected as well.
> 
> You should be able to resolve this issue by deleting `/var/lib/clamav/daily.cld` and then trying again.
> 
> Side note: There are some other issues with freshclam made apparent by the recent major update to daily.cvd and main.cvd (the issue causing this `cdiff_apply: lseek(desc, -350, SEEK_END) failed` error message). We will have to fix the rest of those issues in 0.103.4 & 0.104.1, after the 0.104.0 release.

I have had this same issue with version 0.103.3 on fedora 34. Your suggestion of deleting the daily.cld solved my issue. I will include my specific information below.

Fedora 34
Kernel 5.13.12-200.fc34.x86_64
Messages pulled from a freshclam --show-progress command.
daily.cvd updated (version: 26276, sigs: 1968691, f-level: 90, builder: raynman)
daily database available for update (local version: 26276, remote version: 26278)
